crates/sui-core/src/execution_engine.rs

```diff
@@ -123,6 +123,15 @@ fn execute_transaction<S: BackingPackageStore + ParentSync>(
         // TODO: Since we require all mutable objects to not show up more than
         // once across single tx, we should be able to run them in parallel.
         for single_tx in transaction_data.kind.into_single_transactions() {
+            if single_tx.uses_vm() {
+                // Reserve gas for this VM execution
+                // Reserving only done once for the whole batch
```

Comment:
I don't understand how this is done only once for the whole back. It looks like it is done for each txn in the batch?

Reply:
Right this was something I'm not entirely clear on.
First I'll define a sub_tx as the individual TXes in a batch.
 
There are two ways to think about this topic:
1. We use the same pool for the whole batch because we want this flat fee to limit the total time a submitted transaction can use the VM
2. We use a new pool for each sub_tx, but this means one can hog resources and stall other TXes by having multiple items in batch.

(1) is valid if we want to constrain total VM usage in a batch
(2) is valid if we don't want to constrain total VM usage in a batch. Instead we rely on the MAX_TX_GAS to do that since a batch cannot use more than MAX_TX_GAS anyways.

which approach do you think we want?

Reply:
I'm not quite sure I understand (1) or (2). So I'll restate where I'm coming from in hopes that helps you reframe this.

To me, it makes the most sense to charge the flat fee for each transaction in the batch. Each transaction must have CC <= FF. 
I don't understand this point about hogging resources or total time. None of these seem like considerations that are special for batches. As far as I understand it, batches are just for reducing any cost in network overhead.

Reply:
logic has been simplified